src/json-patch.ts

```diff
@@ -257,30 +270,120 @@ module jsonpatch {
   }
 
   /**
-   * Apply a json-patch operation on an object tree
-   * Returns an array of results of operations.
-   * Each element can either be a boolean (if op == 'test') or
-   * the removed object (operations that remove things)
-   * or just be undefined
+  * Escapes a json pointer path
+  * @param path The raw pointer
+  * @return the Escaped path
+  */
+  export function escapePath(path: string): string {
+    if (path.indexOf('/') === -1 && path.indexOf('~') === -1) return path;
+    return path.replace(/~/g, '~0').replace(/\//g, '~1');
+  }
+  /**
+   * Unescapes a json pointer path
+   * @param path The escaped pointer
+   * @return The unescaped path
+   */
+  export function unEscapePath(path: string): string {
+    return path.replace(/~1/g, '/').replace(/~0/g, '~');
+  }
+
+  /**
+   * Retrieves a value from a JSON document by a JSON pointer.
+   * Returns the value.
+   *
+   * @param document The document to get the value from
+   * @param pointer an escaped JSON pointer
+   * @return The retrieved value
+   */
+  export function getValueByPointer(document, pointer) {
+    let pathSegments = [];
+    if (pointer) { //empty string evaluates to false
+      pathSegments = pointer.substring(1).split(/\//).map(jsonpatch.unEscapePath);
+    }
+    for (let i = 0; i < pathSegments.length; i++) {
+      var subPropertyKey = pathSegments[i];
+      if (!(subPropertyKey in document)) {
```

Comment:
Went back for `in`, keys need to be preserved even if they have undefined values.